library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -760,6 +852,28 @@ public void onLoadCanceled(
     }
   }
 
+  private void discardAfterInterrupt() {
+    int lastChunkIndex = mediaChunks.size() - 1;
+    int preferredQueueSize = lastChunkIndex;
+    for (int i = lastChunkIndex; i >= 0; i--) {
```

Comment:
This logic always calculates preferredQueueSize = lastChunkIndex I think, because there can only be one canceled chunk. 

If you implement my other comment about moving all logic inside getPreferredQueueSize/evaluateQueueSize, then you can just return queueSize-1 if you want to discard the last, currently loading chunk only.